### PR TITLE
#1475 react-docgen node script

### PR DESCRIPTION
**Pull request by fernandinand**
The script takes as input a given plugins/components folder and a target output folder for generating a MD documentation file for each doc-gen ready jsx file recursively.
e.g. ```node genDocs web/client/components docs/developer-guide/reference```

### CODE REVIEW

File: genDocs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  // FIXME: static output folder?
+  fs.writeFile('docs/developer-guide/reference/index.md', tocContent, (err) => {
+      if (err) throw err;
+    });
+}
+
+function checkExt(file, ext) {
+    if (path.extname(file) == ext) {
+        //filelist.concat(file);
+        const docgen = spawn('react-docgen', [ file ]);
```
Comment (mbarto) [inline genDocs]: Why do we spawn an external process, we already have the react parser in memory with require, we should use that (by the way, this doesn't work on Windows)?

File: genDocs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    }
+}
+
+function buildDocs(doc, file) {
+  // Extract filename as component name
+  var name = getComponentName(file);
+  // Use fs to write the markdown to dist
+  var md = docsToMarkdown(doc, name);
+  // FIXME: static output folder?
+  fs.writeFile('docs/developer-guide/reference/' + name +'.md', md, (err) => {
```
Comment (mbarto) [inline genDocs]: If reference folder does not exist (actual case) it breaks, we should create not existing folders

File: package.json [hunk 3]
```diff
@@ -114,6 +117,8 @@
     "react-copy-to-clipboard": "4.1.0",
     "react-dnd": "2.1.3",
     "react-dnd-html5-backend": "2.1.2",
+    "react-docgen": "2.13.0",
```
Comment (mbarto) [inline package.json]: these should go in the devDependencies section

**Comment by offtherailz** [comment]
We decided to use docma instead, going to close.

**Pull request closed without merging**